kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/converter/ZipFileRecordConverter.kt

```
@@ -0,0 +1,84 @@
+import java.util.zip.ZipInputStream
+
+/**
+ * Abstract Zip file converter. Implementing classes should provide specific source-type and compatible
+ * DataProcessors that can process each entry in the Zip file.
+ */
+abstract class ZipFileRecordConverter(sourceType: String, listOfDataProcessors: List<DataProcessor>) : RecordConverter(sourceType) {
+
+    private var processors: Map<String, DataProcessor> = listOfDataProcessors.map { it.schemaType to it }.toMap()
+
+    override fun processData(contents: ContentsDTO, inputStream: InputStream, record: RecordDTO, timeReceived: Double): List<TopicData> {
+        log(LogLevel.INFO, "Retrieved file content from record id ${record.id} and filename ${contents.fileName}")
+        val convertedTopicData = mutableListOf<TopicData>()
+        try {
+            val zippedInput = ZipInputStream(inputStream)
+            zippedInput.use {
+                generateSequence { it.nextEntry }
+                        .forEach { zippedEntry ->
+                            val entryName = zippedEntry.name.trim()
+                            logger.debug("Processing entry $entryName from record ${record.id}")
+                            convertedTopicData.addAll(processContent(object : FilterInputStream(zippedInput) {
+                                @Throws(IOException::class)
+                                override fun close() {
+                                    logger.debug("Closing entry $entryName")
+                                    zippedInput.closeEntry()
+                                }
+                            }, entryName, timeReceived))
+                        }
+            }
+            convertedTopicData.last().endOfFileOffSet = true
+        } catch (exe: IOException) {
+            log(LogLevel.ERROR, "Failed to process zipped input from record ${record.id}", exe)
+        } catch (exe: Exception) {
+            log(LogLevel.ERROR, "Could not process record ${record.id}", exe)
```

Why not rethrow the error? Because the processing will be incomplete right?

---

Yes. It should be thrown and updated as FAILED to the upload-backend-service. I am working on a separate branch to generally improve the error/exception handling and send updates to backend. I can re-throw it for now. But I want to think of a better solution for error handling